src/LinAlg/matrix.h

```diff
@@ -753,6 +755,32 @@ namespace utl //! General utility classes and functions.
       n[2] = n_3;
     }
 
+    //! \brief Constructor fill matrix from istream.
+    matrix3d(const std::string& fname)
+    {
+      std::ifstream is(fname, std::ios::in);
+      for (size_t i=0; i<3; ++i)
+        is >> n[i];
+      double num;
+      elem.clear();
+      while (is >> num)
+        elem.push_back(num);
+      if (n[0]*n[1]*n[2] != elem.size())
+        std::cerr << "matrix3d: error reading from file, expected size "
+                  << n[0]*n[1]*n[2] << " but got " << elem.size() << std::endl;
+    }
+    //! \brief Dump contents to output stream.
+    void dump(const std::string& fname, size_t precision = 16)
+    {
+      std::ofstream os(fname);
+      os << std::setprecision(precision);
+      os << n[0] << std::endl << n[1] << std::endl << n[2] << std::endl;
+      for (size_t k=1; k<=n[2]; ++k)
+        for (size_t j=1; j<=n[1]; ++j)
+          for (size_t i=1; i<=n[0]; ++i)
+            os << this->operator()(i,j,k) << "\n";
+    }
+
     //! \brief Resize the matrix to dimension \f$n_1 \times n_2 \times n_3\f$.
```

Comment:
I understand your POV, I decided against supplying a stream so that I could call dump in one line of code. The << operator has already been implemented so an alternative would be to implement the corresponding >> instead.